datetime/src/time.rs

```diff
@@ -100,29 +106,66 @@ fn time(args: Args) -> Result<(), Box<dyn std::error::Error>> {
             elapsed.as_secs_f64(),
             user_time,
             system_time
-        )?;
+        ).map_err(|_| TimeError::ExecTime)?;
     } else {
         writeln!(
             io::stderr(),
             "Elapsed time: {:.6} seconds\nUser time: {:.6} seconds\nSystem time: {:.6} seconds",
             elapsed.as_secs_f64(),
             user_time,
             system_time
-        )?;
+        ).map_err(|_| TimeError::ExecTime)?;
     }
 
-    std::process::exit(status.code().unwrap_or(1));
+    Ok(())
+}
+
+enum Status {
+    Ok,
+    TimeError,
+    UtilError,
+    UtilNotFound,
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
+impl Status {
+    fn exit(self) -> ! {
+        let res = match self {
+            Status::Ok => 0,
+            Status::TimeError => 1,
+            Status::UtilError => 126,
+            Status::UtilNotFound => 127,
+        };
+
+        std::process::exit(res)
+    }
+}
+
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {   
     // parse command line arguments
-    let args = parse()?;
+    let args = match parse() {
+        Ok(res) => res,
+        Err(err) => {
+            eprintln!("{}", err);
+            Status::UtilNotFound.exit()
```

Comment:
Why is UtilNotFound exit here? With parsing errors we should exit with codes 1-125